src/components/icon/icon.style.ts

```diff
@@ -57,9 +57,24 @@ export interface StyledIconInternalProps {
   theme?: ThemeObject;
 }
 
+const sizeValues: Record<BgSize | FontSize, number> = {
+  "extra-small": 1,
+  small: 2,
+  medium: 3,
+  large: 4,
+  "extra-large": 5,
+};
+
 function adjustIconBgSize(fontSize?: FontSize, bgSize?: BgSize) {
-  if (fontSize && fontSize !== "small") {
-    return iconConfig.backgroundSize[fontSize];
+  if (fontSize && bgSize && fontSize !== "small") {
```

Comment:
**question**: do we need the `fontSize !== "small"` check here at all any more? I think prior to your change that was necessary in order to keep the background as per the `bgSize` prop when `fontSize` was small, but the logic you have here now I don't think needs any exception. (If `fontSize` is `"small"` then `bgSizeValue < fontSizeValue` can never be true so we get `iconConfig.backgroundSize[bgSize]` returned, which is correct and the exact same as we'd get from line 80.

Not a huge deal but I find it more readable without the extra condition, which makes you stop and think "hang on, why do we have a special case for small font-size"? 😅

Reply:
The above comment was made before I noticed that `extra-small` was a possible value for `bgSize` (but not `fontSize`)! In this case we can have `bgSizeValue < fontSizeValue` with `fontSizeValue="small"` - which means what you've got here would ignore that case. So if the consumer provides `<Icon fontSize="small" bgSize="extra-small" />`, your code here would end up with an `extra-small` background size, whereas my suggestion would result in `small`. I think `small` is correct here as otherwise the background is smaller than the font, which I thought is what we're trying to avoid?

You can actually see this case in the icons you added to the `All` story, here's a screenshot highlighting the problematic size combination (small icon in extra-small background): 
![image](https://github.com/Sage/carbon/assets/25186376/3318a797-7813-44ed-8635-7cf4e805e393)
Maybe it doesn't look so bad but I'll leave that with you and/or Harpal to judge!

Reply:
Harpal has suggested removing the "extra-small" `bgSize`, so I've deprecated it and added a warning for this. 
This prop was only being used in components internally so they have been adjusted slightly to avoid any large visual changes.